### PR TITLE
frontend/fix: Ride OTP View Fix

### PR DESCRIPTION
**Pull request by PraveenGongada**
## Type of Change

- [x] Bugfix
- [ ] New feature
- [ ] Enhancement
- [ ] Refactoring
- [ ] Dependency updates

## Description

Made Ride OTP attached to bottom sheet

### CODE REVIEW

File: Frontend/ui-customer/src/Components/DriverInfoCard/View.purs [hunk 1]
```diff
@@ -135,6 +150,301 @@ driverInfoViewSpecialZone push state =
...
+  , width MATCH_PARENT
+  , cornerRadius 16.0
+  , margin $ Margin 0 4 4 4
+  , background $ colorForWaitTime state
+  , gravity CENTER_VERTICAL
+  , accessibility DISABLE
+  ][ textView $ 
+     [ height $ WRAP_CONTENT 
+     , width $ V $ 60
+     , padding $ Padding 0 4 0 6
```
Comment (kavya-shree-s) [inline Frontend/ui-customer/src/Components/DriverInfoCard/View.purs:344]: use paddingVertical

File: Frontend/ui-customer/src/Components/DriverInfoCard/View.purs [hunk 2]
```diff
@@ -135,6 +150,301 @@ driverInfoViewSpecialZone push state =
...
+     , accessibilityHint $ waitTimeHint state
+     , accessibility ENABLE 
+     , text $ state.data.waitingTime
+     , color Color.black900
+     , gravity CENTER
+     , singleLine true
+     ] <> FontStyle.body22 TypoGraphy
+  ]
+
+waitTimeHint :: DriverInfoCardState -> String
```
Comment (kavya-shree-s) [inline Frontend/ui-customer/src/Components/DriverInfoCard/View.purs:354]: Could be wrote in better way

Comment (kavya-shree-s) [reply Frontend/ui-customer/src/Components/DriverInfoCard/View.purs:354]:
```
waitTimeHint :: DriverInfoCardState -> String
waitTimeHint state =
  let
    timePrefix = if state.props.currentSearchResultType == QUOTES
                  then "OTP Expires in: "
                  else "Wait Time: "
    timeParts = STR.split (STR.Pattern ":") state.data.waitingTime
  in
    case timeParts of
      [minutes, seconds] ->
        let
          min = trim minutes
          sec = trim seconds
        in
          if min /= "00"
            then min <> " Minutes and " <> sec <> " Seconds"
            else sec <> " Seconds"
      _ -> "
```

File: Frontend/ui-customer/src/Components/MessagingView/Common/View.purs [hunk 3]
```diff
@@ -365,7 +358,7 @@ quickReplyItem push state item idx =
   , visibility $ boolToVisibility $ not $ DS.null message
   , accessibilityHint $ (getMessageFromKey item $ "EN_US") <> ": Button : Select to send message to driver"
   , onClick (\action -> do
-                when (not $ DS.null state.lastReceivedMessage.sentBy) $ do void $ startTimer 3 ("ChatNotificationRemoval" <> (show $ state.timerCounter)) "3" push state.messageExpiryAction
+                when (not $ DS.null state.lastReceivedMessage.sentBy) $ do void $ startTimer 3 ("ChatNotificationRemoval" <> (show $ state.timerCounter)) "1" push state.messageExpiryAction
```
Comment (kavya-shree-s) [inline Frontend/ui-customer/src/Components/MessagingView/Common/View.purs:361]: @PraveenGongada Can we get this 3 sec from config

File: Frontend/ui-customer/src/Screens/RideBookingFlow/HomeScreen/Controller.purs [hunk 4]
```diff
@@ -3011,9 +3015,12 @@ getInfoCardPeekHeight :: HomeScreenState -> Int
 getInfoCardPeekHeight state = 
   let bottomSheetLayout = (runFn1 getLayoutBounds $ getNewIDWithTag (if state.data.currentSearchResultType == QUOTES then "driverInfoViewSpecialZone" else "driverInfoView"))
       brandingBanner = runFn1 getLayoutBounds $ getNewIDWithTag "BrandingBanner"
+      actionsView = runFn1 getLayoutBounds $ getNewIDWithTag "DriverInfoCardActionView"
       pixels = runFn1 getPixels FunctionCall
       density = (runFn1 getDeviceDefaultDensity FunctionCall)/  defaultDensity
-      currentPeekHeight = bottomSheetLayout.height + if state.data.config.driverInfoConfig.footerVisibility then brandingBanner.height else 0
+      currentPeekHeight = if bottomSheetLayout.height == 0 || actionsView.height == 0
+                          then 0
+                          else bottomSheetLayout.height + if state.data.config.driverInfoConfig.footerVisibility then brandingBanner.height else 0 + actionsView.height
```
Comment (kavya-shree-s) [inline Frontend/ui-customer/src/Screens/RideBookingFlow/HomeScreen/Controller.purs:3023]: In else case can we just return `actionsView.height` ?

File: Frontend/ui-customer/src/Screens/RideBookingFlow/HomeScreen/Controller.purs [hunk 5]
```diff
@@ -886,15 +886,13 @@ data Action = NoAction
...
-            | WaitingInfo
-            | ShareRide
             | ScrollStateChanged String
             | RemoveNotification
             | MessageDriver
             | SendQuickMessage String
             | MessageExpiryTimer Int String String
             | NotificationAnimationEnd
-            | RideSupport
+            | ShareRide
```
[the remark this answers is not in the document]
Comment (PraveenGongada) [reply Frontend/ui-customer/src/Screens/RideBookingFlow/HomeScreen/Controller.purs:895]: yes, for safety

File: Frontend/ui-customer/src/Screens/RideBookingFlow/HomeScreen/Controller.purs [hunk 6]
```diff
@@ -3011,9 +3015,12 @@ getInfoCardPeekHeight :: HomeScreenState -> Int
 getInfoCardPeekHeight state = 
   let bottomSheetLayout = (runFn1 getLayoutBounds $ getNewIDWithTag (if state.data.currentSearchResultType == QUOTES then "driverInfoViewSpecialZone" else "driverInfoView"))
       brandingBanner = runFn1 getLayoutBounds $ getNewIDWithTag "BrandingBanner"
+      actionsView = runFn1 getLayoutBounds $ getNewIDWithTag "DriverInfoCardActionView"
```
[the remark this answers is not in the document]
Comment (PraveenGongada) [reply Frontend/ui-customer/src/Screens/RideBookingFlow/HomeScreen/Controller.purs:3018]: driverInfoView is being for driver details

File: Frontend/ui-customer/src/Screens/RideBookingFlow/HomeScreen/Controller.purs (hunk 4 above)
[the remark this answers is not in the document]
Comment (PraveenGongada) [reply Frontend/ui-customer/src/Screens/RideBookingFlow/HomeScreen/Controller.purs:3023]: The `0` belongs to this check `if state.data.config.driverInfoConfig.footerVisibility then brandingBanner.height else 0`, updated the code

Comment (PraveenGongada) [reply Frontend/ui-customer/src/Screens/RideBookingFlow/HomeScreen/Controller.purs:3023]: The `0` is for before check, update now

**Review by kavya-shree-s — APPROVED** [review]

**Pull request merged**